plugins/calibrecompanion.koplugin/main.lua

```diff
@@ -315,11 +315,52 @@ function CalibreCompanion:sendBooklists(arg)
     DEBUG("SEND_BOOKLISTS", arg)
 end
 
+-- TODO: move to more central location
+local lfs = require("libs/libkoreader-lfs")
+
+-- We make the simplifying assumption that no-one will want to store books with
+-- a literal \ in the filename.
+-- FIXME: can calibre-on-windows even send paths with \ in them? If not we don't
+-- need to worry about this.
+local function normalize_path(path)
+    return (path:gsub('\\', '/'):gsub('/+', '/'))
+end
+
+local function path_exists(path)
+    return lfs.attributes(path, "mode") ~= nil
+end
+
+-- strip the trailing element from path
+-- note: dirname foo/bar and dirname foo/bar/ are both foo
+-- dirname / is /
+-- dirname foo is .
+local function dirname(path)
+    path = normalize_path(path):gsub("[^/]+/?$", "")
+    if path == "" then
+        return "."
+    end
+    return path
+end
+
+-- as `mkdir -p`
+-- unlike lfs.mkdir(), does not error if the directory already exists.
+local function mkdirs(path)
```

Comment:
This probably makes more sense in util. Maybe `util.makeDir()`?

Reply:
Moved to `util.makeDirs()`